packages/sdk/typescript/human-protocol-sdk/test/escrow.test.ts

```
@@ -0,0 +1,1368 @@
+
+      await expect(escrowClient.getResultsUrl(escrowAddress)).rejects.toThrow(
+        ErrorInvalidEscrowAddressProvided
+      );
+    });
+
+    test('should throw an error if an hasEscrow return false', async () => {
+      const escrowAddress = ethers.constants.AddressZero;
+
+      escrowClient.escrowFactoryContract.hasEscrow.mockReturnValue(false);
+
+      await expect(escrowClient.getResultsUrl(escrowAddress)).rejects.toThrow(
+        ErrorEscrowAddressIsNotProvidedByFactory
+      );
+    });
+
+    test('should successfully getResultsUrl escrow', async () => {
+      const escrowAddress = ethers.constants.AddressZero;
+      const url = FAKE_URL;
+
+      escrowClient.escrowFactoryContract.hasEscrow.mockReturnValue(true);
+      escrowClient.escrowContract.finalResultsUrl.mockReturnValue(url);
+
+      const finalResultsUrl = await escrowClient.getResultsUrl(escrowAddress);
+
+      expect(finalResultsUrl).toEqual(url);
+      expect(
+        escrowClient.escrowContract.finalResultsUrl
+      ).toHaveBeenCalledWith();
+    });
+
+    test('should throw an error if the getResultsUrl fails', async () => {
+      const escrowAddress = ethers.constants.AddressZero;
+
+      escrowClient.escrowFactoryContract.hasEscrow.mockReturnValue(true);
+      escrowClient.escrowContract.finalResultsUrl.mockRejectedValueOnce(
+        new Error()
+      );
+
+      await expect(escrowClient.getResultsUrl(escrowAddress)).rejects.toThrow();
+
+      expect(
+        escrowClient.escrowContract.finalResultsUrl
+      ).toHaveBeenCalledWith();
+    });
+  });
+
+  describe('getTokenAddress', () => {
+    test('should throw an error if escrowAddress is an invalid address', async () => {
+      const escrowAddress = FAKE_ADDRESS;
+
+      await expect(escrowClient.getTokenAddress(escrowAddress)).rejects.toThrow(
+        ErrorInvalidEscrowAddressProvided
+      );
+    });
+
+    test('should throw an error if an hasEscrow return false', async () => {
```

fix writing:
'should throw an error if hasEscrow returns false'